File: Shells/Java/src/examples/LocalSolverExample.java
package examples;

import sudoku.SudokuBoardGenerator;
import sudoku.SudokuFile;
import cspSolver.LocalSolver;

public class LocalSolverExample {

	public static void main(String[] args)
	{
		SudokuFile sf = SudokuBoardGenerator.generateBoard(16, 4, 4, 25);
		LocalSolver solver = new LocalSolver(sf);
		
		Thread t1 = new Thread(solver);
		try
		{
			t1.start();
			t1.join(60000);
			if(t1.isAlive())
			{
				t1.interrupt();
			}
		}catch(InterruptedException e)
		{
		}
		
		if(solver.hasSolution())
		{
			solver.printSolverStats();
			System.out.println(solver.getSolution());	
		}
		
		else
		{
			System.out.println("Failed to find a solution");
		}
		
	}
}

File: Sudoku_Cpp_Shell/examples/LocalSolverExample.cpp
#include<iostream>

#include"../sudoku/SudokuBoardGenerator.hpp"
#include"../sudoku/SudokuFile.hpp"
#include"../cspSolver/LocalSolver.hpp"

int main(){
    SudokuFile sf = generateBoard(9,3,3,5);
    std::cout<<"Here is a randomly generated board:"<<std::endl;
	std::cout<<sf.toString()<<std::endl;
	LocalSolver solver = LocalSolver(sf);
    solver.solve();
    if(solver.hasSolution()){
        solver.printSolverStats();
    	std::cout<<solver.getSolution().toString()<<std::endl;
	} else {
        std::cout<<"Failed to find a solution"<<std::endl;
    }
    return 0;
}
